File: 01_projects/code/python-playwright/cbaps-dawms/robust/dawms/pages/dawms_dashboard_page.py
from playwright.sync_api import Page
from loguru import logger

class DAWMSDashboardPage:
    def __init__(self, page: Page, pw):
        self.page = page
        self.pw = pw
        self.intake_btn = page.locator("button:has-text('Submission Intake')")
        self.pw.wait_for_selector("button:has-text('Submission Intake')")
        logger.info("✅ DAWMS Dashboard loaded")
    
    def go_to_submission_intake(self):
        from .submission_intake_page import SubmissionIntakePage
        self.pw.click_element(self.intake_btn)
        logger.info("➡️ Submission Intake")
        return SubmissionIntakePage(self.page, self.pw)

File: 01_projects/code/java-playwright/cbaps-dawms/robust/src/main/java/com/playwright/dawms/pages/DAWMSDashboardPage.java
package com.playwright.dawms.pages;
import com.microsoft.playwright.*;
import com.playwright.cbaps.library.EnhancedPlaywrightManager;
import org.slf4j.*;

public class DAWMSDashboardPage {
    private static final Logger log = LoggerFactory.getLogger(DAWMSDashboardPage.class);
    private Page page;
    private EnhancedPlaywrightManager pwm;
    private Locator submissionIntakeBtn;
    
    public DAWMSDashboardPage(Page page, EnhancedPlaywrightManager pwm) {
        this.page = page; this.pwm = pwm;
        this.submissionIntakeBtn = page.locator("button:has-text('Submission Intake')");
        pwm.waitUntilElementVisible("button:has-text('Submission Intake')");
    }
    
    public SubmissionIntakePage goToSubmissionIntake() {
        pwm.clickElement(submissionIntakeBtn);
        return new SubmissionIntakePage(page, pwm);
    }
}
